meta/autoid/autoid.go

```diff
@@ -253,6 +261,49 @@ func (alloc *allocator) rebase4Signed(tableID, requiredBase int64, allocIDs bool
 	return nil
 }
 
+// rebase4Sequence won't alloc batch immediately, cause it won't cache value in allocator.
+func (alloc *allocator) rebase4Sequence(tableID, requiredBase int64) error {
+	var newBase, newEnd int64
+	startTime := time.Now()
+	err := kv.RunInNewTxn(alloc.store, true, func(txn kv.Transaction) error {
+		m := meta.NewMeta(txn)
+		currentEnd, err := getAutoIDByAllocType(m, alloc.dbID, tableID, alloc.allocType)
+		if err != nil {
+			return err
+		}
+		if alloc.sequence.Increment > 0 {
+			if currentEnd >= requiredBase {
+				newBase = currentEnd
+				newEnd = currentEnd
+				// Required base satisfied, we don't need to update KV.
+				return nil
+			}
+		} else {
+			if currentEnd <= requiredBase {
+				newBase = currentEnd
+				newEnd = currentEnd
+				// Required base satisfied, we don't need to update KV.
+				return nil
+			}
+		}
+
+		// If we don't want to allocate IDs, for example when creating a table with a given base value,
+		// We need to make sure when other TiDB server allocates ID for the first time, requiredBase + 1
+		// will be allocated, so we need to increase the end to exactly the requiredBase.
+		newBase = requiredBase
+		newEnd = requiredBase
+		_, err = generateAutoIDByAllocType(m, alloc.dbID, tableID, newEnd-currentEnd, alloc.allocType)
+		return err
+	})
+	// TODO: sequence metrics
+	metrics.AutoIDHistogram.WithLabelValues(metrics.TableAutoIDRebase, metrics.RetLabel(err)).Observe(time.Since(startTime).Seconds())
+	if err != nil {
+		return err
+	}
+	alloc.base, alloc.end = newBase, newEnd
```

Comment:
Are `allocator.base` and `allocator.end` useless in the case of allocator with `SequenceType`?

Reply:
yes.

Reply:
So why bother setting it?

Reply:
done